src/packages/engine-core/src/log.ts

```diff
@@ -26,6 +20,66 @@ export interface RustError {
   backtrace: string
 }
 
+export function getMessage(log: string | RustLog | RustError | any): string {
+  if (typeof log === 'string') {
+    return log
+  } else if (isRustError(log)) {
+    return getBacktraceFromRustError(log)
+  } else if (isRustLog(log)) {
+    return getBacktraceFromLog(log)
+  }
+
+  return JSON.stringify(log)
+}
+
+export function getBacktraceFromLog(log: RustLog): string | null {
```

Comment:
Looks like these functions could be moved to sdk later

Reply:
Not sure yet, as this is query engine specific.